### PR TITLE
Feature/cutomer signin flow

### PR DESCRIPTION
**Pull request by DeepaliAgrwl**
Please review the customer signIn flow  and let me know in case of any concern

### CODE REVIEW

File: core/src/main/java/com/hcl/ecomm/core/services/impl/CartServiceImpl.java [hunk 1]
```diff
@@ -175,4 +194,50 @@ public String applyCoupon(String couponApplied) {
...
+        String token = "";
+        String url = "";
+        JSONObject getCartResponse = new JSONObject();
+        String domainName = loginService.getDomainName();
+
+            token = customerToken;
+            url = schema + "://" + domainName + config.customer_getCart_string() ;
+
+        JsonArray cartItems = null;
+        LOG.info("url : " + url);
```
Comment (rishabh-behl) [inline core/src/main/java/com/hcl/ecomm/core/services/impl/CartServiceImpl.java]: @DeepaliAgrwl : change this to log.debug

File: ui.apps/src/main/content/jcr_root/apps/hclecomm/components/content/button/clientlibs/js/button.js [hunk 2]
```diff
@@ -11,7 +11,8 @@ $(function() {
                 document.querySelector('.logout').addEventListener('click', function(){
                     console.log("on click event listner code ")
```
Comment (rishabh-behl) [inline ui.apps/src/main/content/jcr_root/apps/hclecomm/components/content/button/clientlibs/js/button.js]: @DeepaliAgrwl : dont need console logs here

File: core/src/main/java/com/hcl/ecomm/core/services/impl/CartServiceImpl.java (hunk 1 above)
Comment (DeepaliAgrwl) [reply core/src/main/java/com/hcl/ecomm/core/services/impl/CartServiceImpl.java]: log level has been updated. Please check

File: ui.apps/src/main/content/jcr_root/apps/hclecomm/components/content/button/clientlibs/js/button.js (hunk 2 above)
Comment (DeepaliAgrwl) [reply ui.apps/src/main/content/jcr_root/apps/hclecomm/components/content/button/clientlibs/js/button.js]: I have removed the console statement. Please check.

File: core/src/main/java/com/hcl/ecomm/core/config/MagentoServiceConfig.java [hunk 3]
```diff
@@ -24,6 +24,13 @@
   String customerSignupPath = "/V1/customers";
   String customerSigninPath = "/V1/integration/customer/token";
   String customerProfilePath = "/V1/customers/me";
+    String customerCreateCartPath = "/V1/carts/mine";
```
Comment (rishabh-behl) [inline core/src/main/java/com/hcl/ecomm/core/config/MagentoServiceConfig.java:27]: @DeepaliAgrwl : is there no runmode configuration for this OSGI config

File: core/src/main/java/com/hcl/ecomm/core/config/MagentoServiceConfig.java [hunk 4]
```diff
@@ -24,6 +24,13 @@
...
+    String customerAddToCartPath = "/V1/carts/mine/items";
```
Comment (rishabh-behl) [inline core/src/main/java/com/hcl/ecomm/core/config/MagentoServiceConfig.java:28]: @DeepaliAgrwl : why are there 3 variables with same value customerAddToCartPath , customerGetCartPath and customerAddToCartPath

Comment (DeepaliAgrwl) [reply core/src/main/java/com/hcl/ecomm/core/config/MagentoServiceConfig.java:28]: @rishabh-behl one it for AddToCart and One Is for CreateCart. Variable names are also different.

File: core/src/main/java/com/hcl/ecomm/core/config/MagentoServiceConfig.java (hunk 3 above)
Comment (DeepaliAgrwl) [reply core/src/main/java/com/hcl/ecomm/core/config/MagentoServiceConfig.java:27]: No @rishabh-behl . Right now we don't have any runmode configuration for this magentoServiceConfigs as we discussed that we will have that once we have our dedicated environment ready

File: core/src/main/java/com/hcl/ecomm/core/services/impl/CartServiceImpl.java [hunk 5]
```diff
@@ -53,19 +55,29 @@ public String getServicePath() {
...
     @Override
-    public JsonArray getCartItemsDetails(String cartId) {
-        String token = loginService.getToken();
+    public JsonArray getCartItemsDetails(String cartId, String customerToken) {
+        String token = "";
+        String url = "";
         String domainName = loginService.getDomainName();
+        if(customerToken != null && !customerToken.isEmpty()) {
+            token = customerToken;
+            //createGuestCartPath = config.customer_createCart_string();
```
Comment (rishabh-behl) [inline core/src/main/java/com/hcl/ecomm/core/services/impl/CartServiceImpl.java]: @DeepaliAgrwl : remove comments

File: core/src/main/java/com/hcl/ecomm/core/services/impl/CreateCartServiceImpl.java [hunk 6]
```diff
@@ -47,16 +47,23 @@ public String getEmptyCartPath() {
 	
 
 	@Override
-	public JSONObject createGuestCart() {
+	public JSONObject createCart(String customerToken) {
 		LOG.debug("createGuestCart method start.");
```
Comment (rishabh-behl) [inline core/src/main/java/com/hcl/ecomm/core/services/impl/CreateCartServiceImpl.java]: @DeepaliAgrwl : remove this log.debug from here

File: core/src/main/java/com/hcl/ecomm/core/services/impl/CreateCartServiceImpl.java [hunk 7]
```diff
@@ -47,16 +47,23 @@ public String getEmptyCartPath() {
...
+				createGuestCartPath = config.customer_createCart_string();
+			}
+			else {
+				authToken = loginService.getToken();
+				createGuestCartPath = getEmptyCartPath();
+			}
 			String domainName = getDomainName();
-			String createGuestCartPath = getEmptyCartPath();
 			String url = scheme + "://" + domainName + createGuestCartPath;
 			LOG.info("createGuestCartPath  : " + url);
```
Comment (rishabh-behl) [inline core/src/main/java/com/hcl/ecomm/core/services/impl/CreateCartServiceImpl.java]: @DeepaliAgrwl : change this to log, debug

File: core/src/main/java/com/hcl/ecomm/core/services/impl/CreateCartServiceImpl.java [hunk 8]
```diff
@@ -47,16 +47,23 @@ public String getEmptyCartPath() {
 	
 
 	@Override
-	public JSONObject createGuestCart() {
+	public JSONObject createCart(String customerToken) {
 		LOG.debug("createGuestCart method start.");
 		String scheme = "http";
+		String authToken = "";
+		String createGuestCartPath = "";
```
Comment (rishabh-behl) [inline core/src/main/java/com/hcl/ecomm/core/services/impl/CreateCartServiceImpl.java]: @DeepaliAgrwl : replace createGuestCart to createCart everywhere required

File: core/src/main/java/com/hcl/ecomm/core/services/impl/CartServiceImpl.java [hunk 9]
```diff
@@ -53,19 +55,29 @@ public String getServicePath() {
...
         LOG.info("url : " + url);
 
         CloseableHttpClient httpClient = HttpClients.createDefault();
         HttpGet httpGet = new HttpGet(url);
         httpGet.setHeader("Content-Type", "application/json");
-
-        String bearerToken = "Bearer" + token;
-        String finalToken = bearerToken.replaceAll("\"","");
+        httpGet.setHeader("Authorization", "Bearer " +token);
+        //String bearerToken = "Bearer" + token;
```
Comment (rishabh-behl) [inline core/src/main/java/com/hcl/ecomm/core/services/impl/CartServiceImpl.java]: @DeepaliAgrwl : Remove unwanted comments

File: core/src/main/java/com/hcl/ecomm/core/servlets/UpdateCartItems.java [hunk 10]
```diff
@@ -29,11 +29,12 @@
     protected void doGet(SlingHttpServletRequest request, SlingHttpServletResponse response)
             throws ServletException, IOException {
         String  cartItemResponse = "";
+        String customerToken = request.getHeader("CustomerToken");
```
Comment (rishabh-behl) [inline core/src/main/java/com/hcl/ecomm/core/servlets/UpdateCartItems.java:32]: @DeepaliAgrwl : why are we getting customer token from request header ? where is that saved in UI ?

File: ui.apps/src/main/content/jcr_root/apps/hclecomm/components/content/cartproductdisplay/clientlibs/js/cartproductdisplay.js [hunk 11]
```diff
@@ -123,6 +123,7 @@ localStorage.setItem('productDescription', JSON.stringify(productDescription ))
   };
  	xhttp.open("GET", "/bin/hclecomm/updateCartItems?payload=" +  JSON.stringify(cartItem) , true);
     xhttp.setRequestHeader("Content-Type", "application/json;charset=UTF-8");
+	  xhttp.setRequestHeader("CustomerToken", custToken);
```
Comment (rishabh-behl) [inline ui.apps/src/main/content/jcr_root/apps/hclecomm/components/content/cartproductdisplay/clientlibs/js/cartproductdisplay.js:126]: @DeepaliAgrwl : where is the assignation for  custToken done

File: core/src/main/java/com/hcl/ecomm/core/services/impl/DeleteCartItemServiceImpl.java [hunk 12]
```diff
@@ -39,20 +39,37 @@ public String getGuestCartItemDeletePath() {
...
+				token = loginService.getToken();
+				itemDeletePath = getGuestCartItemDeletePath();
+				itemDeletePath= itemDeletePath.replace("{cartId}", cartId).replace("{itemId}", itemId);
+				url = scheme + "://" + domainName + itemDeletePath;
+			}
+
+			LOG.info("itemDeletePath  : " + url);
 			Integer statusCode;
 			CloseableHttpClient httpClient = HttpClients.createDefault();
 			HttpDelete httppost = new HttpDelete(url);
```
Comment (rishabh-behl) [inline core/src/main/java/com/hcl/ecomm/core/services/impl/DeleteCartItemServiceImpl.java]: @DeepaliAgrwl : why does a class HttpDelete has variable name httppost ?

File: core/src/main/java/com/hcl/ecomm/core/servlets/GetCustomerCartServlet.java [hunk 13]
```diff
@@ -0,0 +1,54 @@
...
+	private static final long serialVersionUID = 4016057296495129474L;
+	private static final Logger LOG = LoggerFactory.getLogger(GetCustomerCartServlet.class);
+
+	@Reference
+	CartService cartService;
+
+	protected void doGet(SlingHttpServletRequest request, SlingHttpServletResponse response)
+			throws ServletException, IOException {
+
+		LOG.info("inside ProductServlet doGET method");
```
Comment (rishabh-behl) [inline core/src/main/java/com/hcl/ecomm/core/servlets/GetCustomerCartServlet.java]: @DeepaliAgrwl : remove this log

File: ui.apps/src/main/content/jcr_root/apps/hclecomm/components/content/checkoutform/clientlibs/js/checkoutform.js [hunk 14]
```diff
@@ -180,7 +183,7 @@ function onSaveNDeliver() {
 			}
 
 			if (userDetails === 'Pin Number' && userDetailsValue) {
-				const pinRegexp = /^[0-9]{5}(?:-[0-9]{4})?$/;
+				const pinRegexp = /^[0-9]{6}(?:-[0-9]{4})?$/;
```
Comment (rishabh-behl) [inline ui.apps/src/main/content/jcr_root/apps/hclecomm/components/content/checkoutform/clientlibs/js/checkoutform.js]: @DeepaliAgrwl : was this regex change expected

**Review by rishabh-behl — COMMENTED** [review]
@DeepaliAgrwl : Pl incorporate comments

File: core/src/main/java/com/hcl/ecomm/core/servlets/UpdateCartItems.java (hunk 10 above)
Comment (DeepaliAgrwl) [reply core/src/main/java/com/hcl/ecomm/core/servlets/UpdateCartItems.java:32]: @rishabh-behl  this token is provided from the ajax call when the call is placed from ui.
Like this:
        xhttp.open("POST", "/bin/hclecomm/shipinfo", true);
	xhttp.setRequestHeader("Content-Type", "application/json;charset=UTF-8");
    	xhttp.setRequestHeader("CustomerToken", custToken);

File: ui.apps/src/main/content/jcr_root/apps/hclecomm/components/content/cartproductdisplay/clientlibs/js/cartproductdisplay.js (hunk 11 above)
Comment (DeepaliAgrwl) [reply ui.apps/src/main/content/jcr_root/apps/hclecomm/components/content/cartproductdisplay/clientlibs/js/cartproductdisplay.js:126]: It Is given on top of the JS file inside document.ready {} function
Like This:
let userData = getUserCookie("hcluser"); 
    if(userData != "") {
         custToken = JSON.parse(userData).customerToken;
    }

File: core/src/main/java/com/hcl/ecomm/core/servlets/GetCustomerCartServlet.java (hunk 13 above)
Comment (DeepaliAgrwl) [reply core/src/main/java/com/hcl/ecomm/core/servlets/GetCustomerCartServlet.java]: removed

File: ui.apps/src/main/content/jcr_root/apps/hclecomm/components/content/checkoutform/clientlibs/js/checkoutform.js (hunk 14 above)
Comment (DeepaliAgrwl) [reply ui.apps/src/main/content/jcr_root/apps/hclecomm/components/content/checkoutform/clientlibs/js/checkoutform.js]: This change was coming after resolving the conflicts for this pull request. Nithesh has added this for the checkout expiry date feature. So it is taken from dev branch only

File: core/src/main/java/com/hcl/ecomm/core/services/impl/DeleteCartItemServiceImpl.java (hunk 12 above)
Comment (DeepaliAgrwl) [reply core/src/main/java/com/hcl/ecomm/core/services/impl/DeleteCartItemServiceImpl.java]: incorporated. Relevant name has been used

File: core/src/main/java/com/hcl/ecomm/core/services/impl/CartServiceImpl.java (hunk 9 above)
Comment (DeepaliAgrwl) [reply core/src/main/java/com/hcl/ecomm/core/services/impl/CartServiceImpl.java]: Done

File: core/src/main/java/com/hcl/ecomm/core/services/impl/CreateCartServiceImpl.java (hunk 8 above)
Comment (DeepaliAgrwl) [reply core/src/main/java/com/hcl/ecomm/core/services/impl/CreateCartServiceImpl.java]: Done

File: core/src/main/java/com/hcl/ecomm/core/services/impl/CreateCartServiceImpl.java (hunk 7 above)
Comment (DeepaliAgrwl) [reply core/src/main/java/com/hcl/ecomm/core/services/impl/CreateCartServiceImpl.java]: Done

File: core/src/main/java/com/hcl/ecomm/core/services/impl/CreateCartServiceImpl.java (hunk 6 above)
Comment (DeepaliAgrwl) [reply core/src/main/java/com/hcl/ecomm/core/services/impl/CreateCartServiceImpl.java]: Done

File: core/src/main/java/com/hcl/ecomm/core/services/impl/CartServiceImpl.java (hunk 5 above)
Comment (DeepaliAgrwl) [reply core/src/main/java/com/hcl/ecomm/core/services/impl/CartServiceImpl.java]: Done

**Pull request merged**